pkg/uhttp/wrapper.go

```diff
@@ -203,16 +203,53 @@ func WithJSONBody(body interface{}) RequestOption {
 	}
 }
 
+func WithFormBody(body string) RequestOption {
+	return func() (io.ReadWriter, map[string]string, error) {
+		var buffer bytes.Buffer
+		_, err := buffer.WriteString(body)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		_, headers, err := WithContentTypeFormHeader()()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return &buffer, headers, nil
+	}
+}
+
 func WithAcceptJSONHeader() RequestOption {
-	return WithHeader("Accept", "application/json")
+	return WithAccept("application/json")
 }
 
 func WithContentTypeJSONHeader() RequestOption {
-	return WithHeader("Content-Type", "application/json")
+	return WithContentType("application/json")
 }
 
 func WithAcceptXMLHeader() RequestOption {
-	return WithHeader("Accept", "application/xml")
+	return WithAccept("application/xml")
+}
+
+func WithContentTypeFormHeader() RequestOption {
+	return WithContentType("application/x-www-form-urlencoded")
+}
+
+func WithContentTypeVndHeader() RequestOption {
+	return WithContentType("application/vnd.api+json")
+}
+
+func WithAcceptVndJSONHeader() RequestOption {
+	return WithAccept("application/vnd.api+json")
```

Comment:
Since all these strings ("application/json", "application/xml", etc) are duplicated in several places, maybe it makes sense to make them constants.

Reply:
Fixed